crates/re_types_builder/src/context.rs

```diff
@@ -0,0 +1,125 @@
+use std::sync::mpsc;
+
+/// Creates a new context.
+///
+/// The [`Context`] can be freely cloned and sent to other threads.
+///
+/// The [`ContextRoot`] should not be sent to other threads.
+pub fn context() -> (ContextRoot, Context) {
+    let (tx, rx) = mpsc::channel();
+    (ContextRoot::new(rx), Context::new(tx))
+}
+
+#[derive(Clone)]
+pub struct Context {
+    errors: mpsc::Sender<anyhow::Error>,
+}
+
+impl Context {
+    fn new(errors: mpsc::Sender<anyhow::Error>) -> Self {
+        Self { errors }
+    }
+
+    pub fn error(&self, error: impl IntoError) {
+        let _ = self.errors.send(error.into_error());
+    }
+}
+
+pub struct ContextRoot {
+    errors: mpsc::Receiver<anyhow::Error>,
+    _not_send: std::marker::PhantomData<*mut ()>,
+}
+
+impl ContextRoot {
+    fn new(errors: mpsc::Receiver<anyhow::Error>) -> Self {
+        Self {
+            errors,
+            _not_send: std::marker::PhantomData,
+        }
+    }
+
+    /// This outputs all errors to stderr and panics if there were any.
+    pub fn panic_on_errors(&self) {
+        let mut errored = false;
+
+        while let Ok(err) = self.errors.try_recv() {
+            errored = true;
+            eprintln!("{err}");
+        }
+
+        #[allow(clippy::manual_assert)] // we don't want the noise of an assert
+        if errored {
+            panic!("Some errors occurred.");
+        }
+    }
+}
+
+const _: () = {
+    trait IsNotSend<T> {
+        fn __() {}
+    }
+
+    type False = ();
+
+    struct True;
+
+    struct Check<T: ?Sized>(T);
+
+    impl<T: ?Sized> IsNotSend<True> for Check<T> {}
+
+    impl<T: ?Sized + Send> IsNotSend<False> for Check<T> {}
+
+    // if this fails with a type inference error,
+    // then `ContextRoot` is `Send`, which it should _not_ be.
```

Comment:
maybe add a comment about why this is needed

Also: we should consider using something like https://docs.rs/static_assertions/latest/static_assertions/macro.assert_not_impl_any.html or https://crates.io/crates/negative-impl/0.1.4 instead of this magical incantation :)

Reply:
Will do, I wanted to enforce that this should exist only on one thread (preferably the main one).

I thought about using static_assertions, but I don't want us to pay the compilation time cost for just one usage of it